spec/unit/config_spec.rb

```diff
@@ -773,6 +773,10 @@ def assert_valid
       before do
         machine.config.instance_variable_get("@keys")[:vm] = vm
         allow(vm).to receive(:box).and_return(box)
+
+        if machine.provider_config.qemu_use_session
+          allow(vm).to receive(:synced_folders).and_return({})
+        end
```

Comment:
Instead of adding an if condition here, I think it might be cleaner to simply add the allow statement into the before block at the top level of `describe '#validate'`, that way if we add anything else to this test area it'll hopefully not cause an issue.

https://github.com/vagrant-libvirt/vagrant-libvirt/pull/1721/files#diff-09ccf4419aa22d880ea02ed88b14262ad6c20e1c02e948280dd7b36b6e9a2a4fR750

Reply:
Yes, the if condition might be redundant.

I'll attempt to add it into the before block at the `describe` level.

Reply:
Hmm, a few things.
1) not sure why only the MAC spec gets tripped up about this.
2) not sure that it should be in the describe level. AFAICT there are also some synced folder validation specs and I am worried we'd then have false-passes in the test suite. Not 100% sure how RSpec would react.

Reply:
ah, yes, maybe just remove the if condition and leave it at the same location.

I suspect it would be safe to use `subject.qemu_use_session = false` in the before at the describe level as the synced_folder tests will set that attribute to the appropriate value while by default it will ensure the main expectation of using defaulting to the system context for the validate tests remains true. Either should work, I'll it to your preference.

Reply:
I just got rid of the condition then. I think it is better than forcing session/system for specific tests.